capstone-frontend/src/components/SearchUserModal.jsx

```diff
@@ -0,0 +1,255 @@
+      searchResults: [],
+      addedUsersTracker: [],
+      addedUsers: [],
+      addedFriends: [],
+      resultsFound: false,
+    }
+  }
+
+  getBooks = async (searchTerm) => {
+
+    this.setState({ fetchingUsers: true })
+
+    let searchResults;
+
+    if (searchTerm === '') {
+      searchResults = [];
+
+      this.setState({ searchResults, fetchingUsers: false, resultsFound: false })
+    }
+    else {
+      searchResults = await fetch(`/api/userSearch?searchTerm=${searchTerm}`)
+        .then(response => response.json())
+        .catch(err => alert(err));
+
+      if (typeof searchResults === 'undefined') {
+        searchResults = [];
+      }
+
+      if (searchResults.length !== 0) {
+        this.setState({ searchResults, fetchingUsers: false, resultsFound: true }) // Guilty
+      }
+      else {
+        this.setState({ searchResults, fetchingUsers: false, resultsFound: false }) // Not Guilty
+      }
+    }
+  }
+
+  handleSearchTermChange = (event) => {
+
+    // resultsFound innocent until proven Guilty
+    this.setState({ searchResults: [], resultsFound: true })
+
+    if (this.state.typingTimeout) {
+      clearTimeout(this.state.typingTimeout);
+    }
+
+    this.setState({
+      searchTerm: event.target.value,
+      typingTimeout: setTimeout(async () => {
+        await this.getBooks(this.state.searchTerm)
+      }, 500)
+    })
+  }
+
+  addUserToAddedUsers = (user) => {
+
+    if (!this.state.addedUsersTracker.includes(user)) {
+      this.setState({ addedUsersTracker: [...this.state.addedUsersTracker, user] })
+    }
+
+    // Rerender
+    this.setState({ addedUsers: [...this.state.addedUsers, user] })
+  }
+
+  addUserToAddedFriends = (user) => {
+    if (!this.state.addedUsersTracker.includes(user)) {
+      this.setState({ addedUsersTracker: [...this.state.addedUsersTracker, user] })
+    }
+
+    // Rerender
+    this.setState({ addedFriends: [...this.state.addedFriends, user] })
+  }
+
+  removeUserFromAddedUsers = (user) => {
+
+    const indexAddedUsers = this.state.addedUsers.indexOf(user);
+    this.state.addedUsers.splice(indexAddedUsers, 1)
+
+    if (!this.state.addedFriends.includes(user)) {
+      const indexAddedUsersTracker = this.state.addedUsersTracker.indexOf(user);
+      this.state.addedUsersTracker.splice(indexAddedUsersTracker, 1)
+    }
+
+    // Rerender
+    this.setState({ addedUsers: this.state.addedUsers, addedUsersTracker: this.state.addedUsersTracker })
+
+  }
+
+  removeUserFromAddedFriends = (user) => {
+
+    const index = this.state.addedFriends.indexOf(user);
+    this.state.addedFriends.splice(index, 1)
+
+    if (!this.state.addedUsers.includes(user)) {
+      const indexAddedUsersTracker = this.state.addedUsersTracker.indexOf(user);
+      this.state.addedUsersTracker.splice(indexAddedUsersTracker, 1)
+    }
+
+    // Rerender
+    this.setState({ addedFriends: this.state.addedFriends, addedUsersTracker: this.state.addedUsersTracker })
+  }
+
+
+
+  render() {
```

Comment:
Generally with this much JSX you can split this up into a few sub-components, though I'll leave that to your judgment here